src/OMSimulatorLib/System.cpp

```diff
@@ -1092,7 +1082,7 @@ oms_status_enu_t oms3::System::setTLMBusGeometry(const oms3::ComRef& cref, const
 oms3::Connection* oms3::System::getConnection(const oms3::ComRef& crefA, const oms3::ComRef& crefB)
 {
   for (auto& connection : connections)
-    if (connection && connection->getSignalA() == crefA && connection->getSignalB() == crefB)
+    if (connection && (connection->getSignalA() == crefA && connection->getSignalB() == crefB || connection->getSignalA() == crefB && connection->getSignalA() == crefB))
```

Comment:
This looks very weird. I think you probably want this,

`if (connection && ((connection->getSignalA() == crefA && connection->getSignalB() == crefB) || (connection->getSignalA() == crefB && connection->getSignalB() == crefA)))`

Reply:
Thanks, I will fix it. 👍